packages/virtual-core/src/index.ts

```diff
@@ -144,15 +144,16 @@ export const observeElementOffset = <T extends Element>(
   }
 
   let offset = 0
-  const fallback = supportsScrollend
-    ? () => undefined
-    : debounce(
-        targetWindow,
-        () => {
-          cb(offset, false)
-        },
-        instance.options.isScrollingResetDelay,
-      )
+  const fallback =
+    instance.options.useScrollendEvent && supportsScrollend
```

Comment:
Let's also use the same logic in observeWindowOffset https://github.com/TanStack/virtual/blob/main/packages/virtual-core/src/index.ts#L192